dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImpl.java

```diff
@@ -2792,20 +2864,35 @@ public Queries getQueries(final Field field, final Date maxDate) {
      */
     public Queries getJsonFieldQueries(final Field field, final Date maxDate) {
         final ContentletJsonAPI contentletJsonAPI = APILocator.getContentletJsonAPI();
-        if (!contentletJsonAPI.isPersistContentAsJson()) {
-            return new NullQueries();
+        if (contentletJsonAPI.isPersistContentAsJson()) {
+            //If we have got this far it means we are a running postgres instance that obviously supports json.
+            final SimpleDateFormat format = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
+            if (DbConnectionFactory.isPostgres()) {
+                final String select = String
```

Comment:
Can we have something like a Interface here and do something like

```

interface XXX {
     getUpdateQuery();
     getAnyOtherQuery();
}

PostgresXXX implements XXX{

}

MsSqlXXX implements XXX {

}

AnyOtherDataBaseXX implements XXX {
}
```

in this way we not need to keep asking by the BataBase all the time

Reply:
I thought about that. but there are cases on which a big SQL is already in place and we need to modify just a small fragment to reach the json field and the query gets built dynamically concatenating those fragments